src/main/java/io/cdap/plugin/scd2/SCD2Plugin.java

```diff
@@ -277,6 +298,30 @@ private void validate(@Nullable Schema actualSchema, FailureCollector failureCol
         failureCollector.addFailure(String.format("The %s field has value %d, it must not be a negative number",
                                                   OFFSET, getEndDateOffset()), null);
       }
+
+      if (!containsMacro(PRESERVE_TARGET) && !containsMacro(IS_TARGET_FIELD)) {
+        if (preserveTarget() && isTargetField == null) {
+          failureCollector.addFailure(
+            String.format("The %s field must be specified if preserve target is set to true.",
+                          IS_TARGET_FIELD), null);
+        }
+
+        if (isTargetField != null) {
+          Schema.Field field = actualSchema.getField(isTargetField);
+          if (field == null) {
+            failureCollector.addFailure(String.format("The %s field '%s' does not exist in input schema.",
+                                                      IS_TARGET_FIELD, isTargetField),
+                                        null).withConfigElement(IS_TARGET_FIELD, isTargetField);
+          } else {
+            Schema schema = field.getSchema();
+            if (!Schema.Type.BOOLEAN.equals(schema.getType())) {
```

Comment:
if the field is a nullable boolean, this leads to a confusing message complaining that the field is not a boolean.

It would be better to explicitly check for a nullable boolean and say that it must be non-null. The alternative is to treat null as false.

In either case, still need to check for nullables.